code/artificial_intelligence/src/ant_colony_optimization/aco_tsp.cpp

```diff
@@ -0,0 +1,170 @@
+/* Part of Cosmos by OpenGenus Foundation
+ * Application of Ant Optimization Algorithm in Travelling Salesman Problem.
+ * Input: N - Number of vertices, followed by adjacency matrix which stores weights of edges
+ *   of complete undirected weighted graph with no loops.
+ * Output: tour thought all unrepeated vertices via the cheapest path and the cost of this tour.
+ */
+#include <iostream>
+#include <vector>
+#include <ctime>
+
+using namespace std;
+
+const double g_rho = 0.6;    // Evaporation constant
+const double g_q = 0.2;      // Constant in equation delta_T_kth_from_i_to_j = q / L_kth
+
+double doubleRand()
+{   
+    // Generate random double within the interval [0.0, 1.0)
+    return double(rand()) / (double(RAND_MAX) + 1.0);
```

Comment:
Is it possible that you could instead use something from the `random` header instead, which is more modern.